### ISSUE TITLE
UnnecessaryAbstractClass false positive for abstractproperties

### ISSUE THREAD

**Issue by WonderCsabo** [arturbosch/detekt#2526]
## Expected Behavior
`UnnecessaryAbstractClass` should allow an abstract class with abstract properties

## Observed Behavior
`UnnecessaryAbstractClass` fails on an abstract class with abstract properties

## Steps to Reproduce
```kotlin
abstract class ChildViewModel : ViewModel() {

    var parentViewModel: BaseViewModel? = null

    public override fun onCleared() {
        viewModelScope.cancel()
        super.onCleared()
    }
}

abstract class BaseTransferWidgetViewModel : ChildViewModel() {

    abstract val items: LiveData<List<RowViewModel<*>>>

    abstract val widgetListMaxSize: Int

    abstract val selectItemCommand: CommandFactory<RowViewModel<*>>

    abstract val showMoreCommand: Command
}
```

`UnnecessaryAbstractClass` fails on `BaseTransferWidgetViewModel`. It allowed it in detekt 1.7.0

## Your Environment

* Version of detekt used: 1.7.1
* Version of Gradle used (if applicable): 6.2.2
* Operating System and version: macOS 10.14.6

**Comment by BraisGabin** [comment]
🤔 `BaseTransferWidgetViewModel` could be an interface safely. But maybe that's a bit opinated: "composition over inheritance". I'm not sure if that's what this rule is looking for.

**Comment by schalkms** [comment]
This is a bug. Thanks for reporting this.
This came in with a bug fix in v1.7.1. Now detekt also considers base classes, when reporting an abstract class without any concrete members. Unfortunately, this case slipped through.
I have a fix in the queue.

**Comment by BraisGabin** [comment]
I think that the rule tells that it could be replaced by an interface and that's true and the reason yout code is reported:

```kotlin
interface ParentClass {
    val type: String
}
```
